### PR TITLE
Remove Pipe Dependency from Action Sequences

### LINKED ISSUE #1057: Why is system/pipe hard coded in action.go?
I think this is a kind of issue that we hard coded "system/pipe" at Line 533 of openwhisk/tools/go-cli/go-whisk-cli/commands/action.go.

If this action is missing, some of our test cases have to fail, because they are unable to create the sequence of actions. All the test cases we have now cannot access the system auth key, which means they are unable to create fixture for this system/pipe action.

I propose to change this way of implementation, at least by replacing it with a parameter/variable(we can keep the system/pipe action as default), so that we can specify a "mocking pipe" action in the test cases.

### PR DESCRIPTION
**Pull request by dubeejw**
- Remove the dependency on the system pipe when creating actions
- Pass action artifacts to the backend as a component field instead of as parameters

Fixes #1057

### CODE REVIEW

File: tools/go-cli/go-whisk-cli/commands/action.go [hunk 1]
```diff
@@ -531,22 +531,23 @@ func parseAction(cmd *cobra.Command, args []string) (*whisk.Action, bool, error)
     currentNamespace := client.Config.Namespace
     client.Config.Namespace = "whisk.system"
 
-    pipeAction, _, err := client.Actions.Get("system/pipe")
+    /*pipeAction, _, err := client.Actions.Get("system/pipe")
```
Comment (rabbah) [inline tools/go-cli/go-whisk-cli/commands/action.go]: just remove this code? it's not coming back.

**Comment by rabbah** [comment]
Can you add the test alluded to here: https://github.com/openwhisk/openwhisk/issues/1057#issuecomment-240458764?

**Comment by dubeejw** [comment]
Will do.

**Comment by dubeejw** [comment]
PG1 214.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/e5cf3c1806b8c1ed2f3dfaba24f9aacedfe81f90.

**Pull request closed without merging**